File: PHP-OOP/src/factory/AbstractPrinterFactory.php
<?php
namespace app\factory;

use app\factory\SystemOutPrinterFactory;

/*
 * contoh class abstract
 * harus dibuat object inheritance kalau mau pake
 */
abstract class AbstractPrinterFactory 
{ 
	public static function getFactory(){ 
		return new SystemOutPrinterFactory(); 
	} 

	public abstract function getPrinter(); 
} 

File: C#-OOP/Factory/AbstractPrinterFactory.cs
using CSharp_OOP.Services;

namespace CSharp_OOP.Factory
{
    public abstract class AbstractPrinterFactory
    {
		public static AbstractPrinterFactory GetFactory()
		{
			return new SystemOutPrinterFactory();
		}

		public abstract IPrinter GetPrinter();
	}
}
